core/src/test/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -110,10 +110,9 @@ public boolean shouldCache(Query query) throws IOException {
 
         QueryShardContext queryShardContext = mock(QueryShardContext.class);
         for (MappedFieldType fieldType : fieldTypes) {
-            IndexFieldData<?> fieldData = fieldType.fielddataBuilder().build(indexSettings, fieldType,
-                new IndexFieldDataCache.None(), circuitBreakerService, mock(MapperService.class));
             when(queryShardContext.fieldMapper(fieldType.name())).thenReturn(fieldType);
-            when(queryShardContext.getForField(fieldType)).thenReturn(fieldData);
+            when(queryShardContext.getForField(fieldType)).then(invocation -> fieldType.fielddataBuilder().build(
```

Comment:
This blew up without fielddata enabled so I made it return when used instead of all the time.